### PR TITLE
replaced use of subprocess with call_binary() where possible (index_node)

### PR DESCRIPTION
**Pull request by William-Hill**

### CODE REVIEW

File: index_node/esg_cog.py [hunk 1]
```diff
@@ -112,30 +111,26 @@ def setup_cog(COG_DIR="/usr/local/cog"):
...
-            pass
-    try:
-        esg_functions.stream_subprocess_output("useradd -r -g cogadmin cogadmin")
-    except SubprocessError, error:
-        logger.debug(error.__dict__["data"]["returncode"])
-        if error.__dict__["data"]["returncode"] == 9:
+        esg_functions.call_binary("useradd", ["-r", "-g", "cogadmin", "cogadmin"])
+    except ProcessExecutionError, err:
+        if err.retcode == 9:
             pass
```
Comment (nathanlcarlson) [inline index_node/esg_cog.py]: Same comment from other reveiw, have a `create_user` function?

File: index_node/solr.py [hunk 2]
```diff
@@ -87,20 +84,20 @@ def start_solr(solr_config_type, port_number, SOLR_INSTALL_DIR="/usr/local/solr"
...
 def solr_status(SOLR_INSTALL_DIR):
     '''Check the status of solr'''
     try:
-        esg_functions.stream_subprocess_output("{SOLR_INSTALL_DIR}/bin/solr status".format(SOLR_INSTALL_DIR=SOLR_INSTALL_DIR))
-    except OSError:
-        pass
-    except SubprocessError:
-        raise("Error checking solr status")
+        esg_functions.call_binary("/usr/local/bin/solr", ["status"])
+    except ProcessExecutionError:
```
Comment (nathanlcarlson) [inline index_node/solr.py]: Is there a difference between `/usr/local/bin/solr` and the `/usr/local/solr/bin/solr` used elsewhere?

Comment (William-Hill) [reply index_node/solr.py]: /usr/local/solr/bin/solr  is the correct path.  The other is a typo.  I'll push a commit to fix

File: index_node/esg_cog.py (hunk 1 above)
Comment (William-Hill) [reply index_node/esg_cog.py]: I'll create a separate issue and add the create_user function

File: index_node/solr.py (hunk 2 above)
Comment (nathanlcarlson) [reply index_node/solr.py]: Sounds good

**Comment by William-Hill** [comment]
Did you ever approve this? @nathanlcarlson

File: index_node/solr.py (hunk 2 above)
Comment (William-Hill) [reply index_node/solr.py]: @nathanlcarlson this PR is ready to go

**Pull request merged**